### DISCUSSION TITLE
SwitchTopicX - how to send MQTT stat instead of cmnd?

### DISCUSSION THREAD

**Discussion by popy2k14** [arendst/Tasmota#16784] [Q&A] [answered]
Hey guys.

Currently i am struggling with an thing and i dont know how to solve it correctly.
I have an tasmota device which has connected two "pet alarm mats" on two gpios and which i am using in HA.
The device is just an mechanical "switch" which i am using.
They are connected like:

- First device switches TX Pin (GPIO) to GND
- Second device switches RX Pin (GPIO) to GND
- pull up's are connected to 3V3 on every input

Electrically they are fine and working as they should.
But i am struggling with cmnd and stat.

The POWER1 is sending an /smarthome/sz/bedsensor/stat/POWER1 on change and i think it's always sending it in an interval.
My configured POWER2 and POWER3 is sending /smarthome/sz/bedsensor/**cmnd**/POWER2 just when the state changes.
**So when HA cores restarts without anyone touches the bed over the day (which is 100% the case), than some automations does'nt trigger.**

Sure i can solve this in HA with an custom sensor which stores the last value, but when the state of the input changes during HA downtime, that's the next issue.

The configuration of tasmota is like is:
- Under device settings -> GPIO1 (TX) to: Switch 2
- Under device settings -> GPIO3 (RX) to: Switch 3
- MQTT topic: /smarthome/sz/%topic%/%prefix%/
- Switchmode2 2              ->  (In 2 Follow)
- Switchmode3 2              ->  (In 3 Follow)
- SwitchTopic2 bedsensor   ->  send MQTT -> do not switch relais
- SwitchTopic3 bedsensor   ->  send MQTT -> do not switch relais

in HA the yaml looks like this:
```
- name: Betten Sensor 2
  state_topic: "/smarthome/sz/bedsensor/cmnd/POWER2"
  qos: 0

- name: Betten Sensor 3
  state_topic: "/smarthome/sz/bedsensor/cmnd/POWER3"
  qos: 0
```

**Any hint's how i can configure tasmota so it sends an stat instead of cmnd and also periodically so when HA is down it get's updated it's sensors in time and prevent state 'unknown'?**

**Comment by barbudor** [comment]
As it happens you want to detach the swiches 2 and 3 from the relay, have read all the options ?
You have chosen the one which makes the switch sending the cmnd message
There are other options such as `SetOption114` or `SwitchMode  15` or Rules that may be more suitable
Actually, Rule allows you to do exactly anything you want including publishing on any topic (using the command Publish) and with Publish2 you will publish on a retained topic which will still be there is HA restart

However having Tasmota to send a stat/POWER message in reaction to a switch state change and not for a power state seems a bit weird to me

**Comment by popy2k14** [comment, marked as answer]
Thanks a lot for your answer.
i have now solved it in HA to retain the state with an temp binary sensor:

```
#needed to retain value on core restart
- platform: template
  sensors:
    betten_sensor:
      friendly_name: "Betten Sensor"
      value_template: >
        {% if states('binary_sensor.betten_sensor_no_retain') in ['unavailable', 'unknown', 'none'] %}
          {{ states('binary_sensor.betten_sensor') }}
        {% else %}
          {{ states('binary_sensor.betten_sensor_no_retain') }}
        {% endif %}  
```
